xdcr/rosmar_xdcr.go

```diff
@@ -61,86 +65,115 @@ func newRosmarManager(ctx context.Context, fromBucket, toBucket *rosmar.Bucket,
 // processEvent processes a DCP event coming from a toBucket and replicates it to the target datastore.
 func (r *rosmarManager) processEvent(ctx context.Context, event sgbucket.FeedEvent) bool {
 	docID := string(event.Key)
-	base.TracefCtx(ctx, base.KeyWalrus, "Got event %s, opcode: %s", docID, event.Opcode)
+	base.TracefCtx(ctx, base.KeySGTest, "Got event %s, opcode: %s", docID, event.Opcode)
 	col, ok := r.toBucketCollections[event.CollectionID]
 	if !ok {
 		base.ErrorfCtx(ctx, "This violates the assumption that all collections are mapped to a target collection. This should not happen. Found event=%+v", event)
 		r.errorCount.Add(1)
 		return false
 	}
+	ctx = base.CorrelationIDLogCtx(context.Background(), fmt.Sprintf("%s->%s\n\t", r.fromBucketKeyspaces[event.CollectionID], col.GetName())) // use context.Background() to drop test information since context is too long
 
 	switch event.Opcode {
 	case sgbucket.FeedOpDeletion, sgbucket.FeedOpMutation:
 		// Filter out events if we have a non XDCR filter
 		if r.filterFunc != nil && !r.filterFunc(&event) {
-			base.TracefCtx(ctx, base.KeyWalrus, "Filtering doc %s", docID)
+			base.TracefCtx(ctx, base.KeySGTest, "Filtering doc %s", docID)
 			r.mobileDocsFiltered.Add(1)
 			return true
 		}
 
 		// Have to use GetWithXattrs to get a cas value back if there are no xattrs (GetWithXattrs will not return a cas if there are no xattrs)
-		_, targetXattrs, toCas, err := col.GetWithXattrs(ctx, docID, []string{base.VvXattrName, base.MouXattrName, base.SyncXattrName})
+		_, targetXattrs, actualTargetCas, err := col.GetWithXattrs(ctx, docID, []string{base.VvXattrName, base.MouXattrName, base.SyncXattrName})
 		if err != nil && !base.IsDocNotFoundError(err) {
 			base.WarnfCtx(ctx, "Skipping replicating doc %s, could not perform a kv op get doc in toBucket: %s", event.Key, err)
 			r.errorCount.Add(1)
 			return false
 		}
 
-		/* full LWW conflict resolution is not implemented in rosmar yet. There is no need to implement this since CAS will always be unique due to rosmar limitations.
+		sourceHLV, sourceMou, nonMobileXattrs, body, err := processDCPEvent(&event)
+		if err != nil {
+			base.WarnfCtx(ctx, "Replicating doc %s, could not get body, hlv, and mou: %s", event.Key, err)
+			r.errorCount.Add(1)
+			return false
+		}
 
-		CBS algorithm is:
+		sourceCas := event.Cas
```

Comment:
Possibly add some comments describing what this block of code is doing, for future reference?